core/templates/services/user-backend-api.service.ts

```diff
@@ -0,0 +1,164 @@
+  'user_bio': string;
+  'subject_interests': string;
+  'can_receive_email_updates': boolean;
+  'can_receive_editor_role_email': boolean;
+  'can_receive_feedback_message_email': boolean;
+  'can_receive_subscription_email': boolean;
+  'subscription_list': List<SubscriptionSummary>;
+}
+interface UrlBackendDict {
+  'login_url': string;
+}
+interface UserCommunityRightsDataBackendDict {
+  'can_review_translation_for_language_codes': boolean;
+  'can_review_voiceover_for_language_codes': boolean;
+  'can_review_questions': boolean;
+}
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private windowRef: WindowRef,
+    private urlInterpolationService: UrlInterpolationService,
+    private urlService: UrlService,
+    private userInfoObjectFactory: UserInfoObjectFactory) {}
+
+    private PREFERENCES_DATA_URL = '/preferenceshandler/data';
+    private USER_COMMUNITY_RIGHTS_DATA_URL = '/usercommunityrightsdatahandler';
+
+    private userInfo = null;
+    private userCommunityRightsInfo = null;
+
+    getUserInfoAsync(): Promise<UserInfo> {
+      return new Promise((resolve, reject) => {
+        if (this.urlService.getPathname() === '/signup') {
+          return resolve(this.userInfoObjectFactory.createDefault());
+        }
+        if (this.userInfo) {
+          return resolve(this.userInfo);
+        }
+        return this.http.get<UserInfoBackendDict>(
+          '/userinfohandler').toPromise().then(
+          (backendDict) => {
+            if (backendDict.user_is_logged_in) {
+              this.userInfo =
+                this.userInfoObjectFactory.createFromBackendDict(
+                  backendDict);
+              return resolve(this.userInfo);
+            } else {
+              return resolve(
+                this.userInfoObjectFactory.createDefault());
+            }
+          });
+      });
+    }
+    getProfileImageDataUrlAsync(): Promise<string> {
+      let profilePictureDataUrl = (
+        this.urlInterpolationService.getStaticImageUrl(
+          AppConstants.DEFAULT_PROFILE_IMAGE_PATH));
+      return this.getUserInfoAsync().then(
+        (userInfo) => {
+          if (userInfo.isLoggedIn()) {
+            return this.http.get<PreferencesBackendDict>(
+              '/preferenceshandler/profile_picture'
+            ).toPromise().then(
+              (backendDict) => {
+                if (backendDict.profile_picture_data_url) {
+                  profilePictureDataUrl =
+                    backendDict.profile_picture_data_url;
+                }
+                return profilePictureDataUrl;
+              });
+          } else {
+            return new Promise((resolve, reject) => {
+              resolve(profilePictureDataUrl);
+            });
+          }
+        });
+    }
+    setProfileImageDataUrlAsync(newProfileImageDataUrl: string):
+      Promise<PreferencesBackendDict> {
+      const putData = {
```

Comment:
can you think of a name that is more descriptive of the content inside, maybe profileImageUrlUpdateData or something along those lines? Otherwise, you can just add the object as the second parameter in line number 133. I think it will be better to just use this directly over in line 133.

Reply:
Done!